GeocodingLambda/integration.test.ts

```diff
@@ -0,0 +1,260 @@
+            longitude: expect.closeTo(coordinate.value.longitude)
+          },
+          placemark: expect.objectContaining(
+            placemark.value
+          )
+        })
+      }
+    )
+
+    it("Should use default coordinates given an unknown address", async () => {
+      const result = await handler(unknownAddressTestLocation.placemark)
+
+      expect(result).toMatchObject({
+        coordinate: {
+          latitude: -90,
+          longitude: 0
+        }
+      })
+
+      expect(result.placemark).toStrictEqual({})
+    })
+  })
+
+  describe("Reverse geocoding (coordinates to address)", () => {
+    it.each(testLocations)(
+      "Should find the address and save a placemark given the coordinates of $name",
+      async ({ coordinate, placemark }) => {
+        const result = await handler(coordinate)
+
+        expect(result).toMatchObject({
+          coordinate: {
+            latitude: expect.closeTo(coordinate.value.latitude),
+            longitude: expect.closeTo(coordinate.value.longitude)
+          },
+          placemark: expect.objectContaining(
+            placemark.value
+          )
+        })
+
+        const cachedResult = await handler(coordinate, mockGeocoding, mockReverseGeocoding)
+        expect(mockGeocoding).toHaveBeenCalledTimes(0)
+        expect(mockReverseGeocoding).toHaveBeenCalledTimes(0)
+
+        expect(cachedResult).toMatchObject({
+          coordinate: {
+            latitude: expect.closeTo(coordinate.value.latitude),
+            longitude: expect.closeTo(coordinate.value.longitude)
+          },
+          placemark: expect.objectContaining(
+            placemark.value
+          )
+        })
+      }
+    )
+
+    it("Should use default address given unknown coordinates", async () => {
+      const result = await handler(unknownAddressTestLocation.coordinate)
+
+      expect(result).toMatchObject({
+        coordinate: {
+          latitude: expect.closeTo(unknownAddressTestLocation.coordinate.value.latitude),
+          longitude: expect.closeTo(unknownAddressTestLocation.coordinate.value.longitude)
+        }
+      })
+
+      expect(result.placemark).toStrictEqual({
+        city: undefined,
+        country: undefined,
+        isoCountryCode: undefined,
+        name: undefined,
+        postalCode: undefined,
+        region: undefined,
+        street: undefined,
+        streetNumber: undefined
+      })
+    })
+  })
+
+  it("should allow multiple identical geocoding requests", async () => {
+    const [result1, result2, result3] =
+      await Promise.all([
+        handler(unknownAddressTestLocation.coordinate),
+        handler(unknownAddressTestLocation.coordinate),
+        handler(unknownAddressTestLocation.coordinate)
+      ])
+
+    expect(result1).toMatchObject(result2)
+    expect(result3).toMatchObject(result2)
+    expect(result1).toMatchObject(result3)
+  })
+
+  afterAll(() => conn.closeConnection())
```

Comment:
Nit: We could make some shared test helper that autocloses the connection after all. Maybe this could just be a global jest setup that gets applied to all targets.